core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -30,6 +30,7 @@ import org.apache.spark.annotation.DeveloperApi
 @DeveloperApi
 class TaskInfo(
     val taskId: Long,
+    /** the index of this task in its TaskSet.  *Not* necessarily the same as the partitionid */
```

Comment:
BTW, at some point a while ago, I attempted to change the naming of these to make it less confusing, but than ran into issues because there are various naming artifacts from Mesos, which uses taskId to refer to our taskAttemptId, so I eventually gave up.  It does seem like we should (separately from this PR) fix all the places the code uses taskId in place of index to say "index" instead for consistency.